### ISSUE TITLE
Enhancement: [prefer-optional-chain] support `if (x != null)` case

### ISSUE THREAD

**Issue by kirkwaiblinger** [typescript-eslint/typescript-eslint#10842]
### Before You File a Proposal Please Confirm You Have Done The Following...

- [x] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3A%22enhancement%3A+plugin+rule+option%22) and found none that match my proposal.
- [x] I have searched the [current rule list](https://typescript-eslint.io/rules/#rules) and found no rules that match my proposal.
- [x] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.

### My proposal is suitable for this project

- [x] I believe my proposal would be useful to the broader TypeScript community (meaning it is not a niche proposal).

### Link to the rule's documentation

https://typescript-eslint.io/rules/prefer-nullish-coalescing/

### Description

Followup to https://github.com/typescript-eslint/typescript-eslint/pull/10137#discussion_r1955076662 - in addition to the `if (x)` case covered by https://github.com/typescript-eslint/typescript-eslint/issues/6309 &rarr; https://github.com/typescript-eslint/typescript-eslint/pull/10137, let's also cover the `if (x != null)` case.

### Fail

```TypeScript
declare const callable: () => void | null;

if (callable != null) {
  callable();
}
```

### Pass

```TypeScript
declare const callable: () => void | null;

callable?.();
```

### Additional Info

Probably best to consider this blocked on https://github.com/typescript-eslint/typescript-eslint/pull/10137
